Library/Homebrew/cmd/install.rb

```diff
@@ -278,48 +278,59 @@ def install
     # formula was found, but there's a problem with its implementation).
     $stderr.puts e.backtrace if Homebrew::EnvConfig.developer?
     ofail e.message
-  rescue FormulaOrCaskUnavailableError => e
-    if e.name == "updog"
+  rescue FormulaOrCaskUnavailableError, Cask::CaskUnavailableError => e
+    # formula name or cask token
+    name = e.try(:name) || e.token
+
+    if name == "updog"
       ofail "What's updog?"
       return
     end
 
     opoo e
-    ohai "Searching for similarly named formulae..."
-    formulae_search_results = search_formulae(e.name)
-    case formulae_search_results.length
-    when 0
-      ofail "No similarly named formulae found."
-    when 1
-      puts "This similarly named formula was found:"
-      puts formulae_search_results
-      puts "To install it, run:\n  brew install #{formulae_search_results.first}"
-    else
-      puts "These similarly named formulae were found:"
-      puts Formatter.columns(formulae_search_results)
-      puts "To install one of them, run (for example):\n  brew install #{formulae_search_results.first}"
-    end
 
-    if (reason = MissingFormula.reason(e.name))
+    reason = MissingFormula.reason(name, silent: true)
+    if !args.cask? && reason
       $stderr.puts reason
       return
     end
 
-    # Do not search taps if the formula name is qualified
-    return if e.name.include?("/")
-
-    taps_search_results = search_taps(e.name)[:formulae]
-    case taps_search_results.length
-    when 0
-      ofail "No formulae found in taps."
-    when 1
-      puts "This formula was found in a tap:"
-      puts taps_search_results
-      puts "To install it, run:\n  brew install #{taps_search_results.first}"
-    else
-      puts "These formulae were found in taps:"
-      puts Formatter.columns(taps_search_results)
-      puts "To install one of them, run (for example):\n  brew install #{taps_search_results.first}"
+    return if name.include?("/")
```

Comment:
We don't seem to get good search results when the tap is specified so we might as well return early.

Reply:
```suggestion
    # We don't seem to get good search results when the tap is specified
    # so we might as well return early.
    return if name.include?("/")
```